components/Buttons/tests/unit/ButtonsThemingTest.swift

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+
+import MaterialComponents.MaterialButtons
+import MaterialComponents.MaterialColorScheme
+import MaterialComponents.MaterialShapeScheme
+import MaterialComponents.MaterialShapeLibrary
+import MaterialComponents.MaterialTypographyScheme
+import MaterialComponentsAlpha.MaterialButtons_Theming
+
+class ButtonsThemingTest: XCTestCase {
+
+  static let disabledOpacity: CGFloat = 0.38
+  static let disabledBackgroundOpacity: CGFloat = 0.12
+  static let inkOpacity: CGFloat = 0.32
+
+  func testContainedTheme() {
+    // Given
+    let button = MDCButton()
+    let scheme = MDCContainerScheme()
+    let colorScheme = MDCSemanticColorScheme(defaults: .material201804)
+    let shapeScheme = MDCShapeScheme(defaults: .material201809)
+    let typographyScheme = MDCTypographyScheme(defaults: .material201804)
+    let baselineCornerRadius: CGFloat = 4
+
+    // When
+    button.applyContainedTheme(withScheme: scheme)
+
+    // Then
+    // Test Colors
+    XCTAssertEqual(button.backgroundColor(for: .normal), colorScheme.primaryColor)
+    XCTAssertEqual(
+        button.backgroundColor(for: .disabled),
+        colorScheme.onSurfaceColor.withAlphaComponent(ButtonsThemingTest.disabledBackgroundOpacity))
+    XCTAssertEqual(button.titleColor(for: .normal), colorScheme.onPrimaryColor)
+    XCTAssertEqual(
+        button.titleColor(for: .disabled),
+        colorScheme.onSurfaceColor.withAlphaComponent(ButtonsThemingTest.disabledOpacity))
+    XCTAssertEqual(button.imageTintColor(for: .normal), colorScheme.onPrimaryColor)
+    XCTAssertEqual(
+        button.imageTintColor(for: .disabled),
+        colorScheme.onSurfaceColor.withAlphaComponent(ButtonsThemingTest.disabledOpacity))
+    XCTAssertEqual(button.inkColor,
+                   colorScheme.onPrimaryColor.withAlphaComponent(ButtonsThemingTest.inkOpacity))
+    // Test shape
+    if let buttonShape = button.shapeGenerator as? MDCRectangleShapeGenerator {
```

Comment:
If the container scheme was initialized with no shape scheme wouldn't the test only need to check for corner radius?